cmd/validate_asserter_config.go

```diff
@@ -0,0 +1,175 @@
+		return err
+	}
+
+	if err := verifyOperationStatuses(
+		networkAllow.OperationStatuses, asserterConfiguration.AllowedOperationStatuses,
+	); err != nil {
+		return err
+	}
+
+	return verifyErrors(networkAllow.Errors, asserterConfiguration.AllowedErrors)
+}
+
+func verifyTimestampStartIndex(networkTsi *int64, assertTsi int64) error {
+	if networkTsi == nil { // This field is optional and defaults to all allowed
+		return nil
+	}
+	if *networkTsi != assertTsi {
+		return fmt.Errorf(
+			"/network/options / asserter-configuration-file timestamp start index mismatch. %d %d",
+			*networkTsi, assertTsi,
+		)
+	}
+
+	return nil
+}
+
+func verifyOperationTypes(networkOt, asserterOt []string) error {
+	if len(networkOt) != len(asserterOt) {
+		return fmt.Errorf(
+			"/network/options / asserter-configuration-file operation types length mismatch %v %v",
+			networkOt, asserterOt,
+		)
+	}
+
+	sort.Strings(networkOt)
+	sort.Strings(asserterOt)
+
+	for i, networkOperationType := range networkOt {
+		asserterOperationType := asserterOt[i]
+		if networkOperationType != asserterOperationType {
+			return fmt.Errorf(
+				"/network/options / asserter-configuration-file operation type mismatch %v %v",
+				networkOperationType, asserterOperationType,
+			)
+		}
+	}
+
+	return nil
+}
+
+func verifyOperationStatuses(networkOs, asserterOs []*types.OperationStatus) error {
+	if len(networkOs) != len(asserterOs) {
+		return fmt.Errorf(
+			"/network/options / asserter-configuration-file operation statuses length mismatch "+
+				"%v %v",
+			networkOs, asserterOs,
+		)
+	}
+
+	sort.Slice(networkOs, func(i, j int) bool {
+		return strings.Compare(networkOs[i].Status, networkOs[j].Status) < 0
+	})
+	sort.Slice(asserterOs, func(i, j int) bool {
+		return strings.Compare(asserterOs[i].Status, asserterOs[j].Status) < 0
+	})
+
+	for i, networkOperationStatus := range networkOs {
+		asserterOperationStatus := asserterOs[i]
+		if !reflect.DeepEqual(networkOperationStatus, asserterOperationStatus) {
+			return fmt.Errorf(
+				"/network/options / asserter-configuration-file operation status mismatch %v %v",
+				networkOperationStatus, asserterOperationStatus,
+			)
+		}
+	}
+
+	return nil
+}
+
+func verifyErrors(networkErrors, asserterErrors []*types.Error) error {
+	if len(networkErrors) != len(asserterErrors) {
+		return fmt.Errorf(
+			"/network/options / asserter-configuration-file errors length mismatch %v %v",
+			networkErrors, asserterErrors,
+		)
+	}
+
+	sort.Slice(networkErrors, func(i, j int) bool {
+		return networkErrors[i].Code < networkErrors[j].Code
+	})
+	sort.Slice(asserterErrors, func(i, j int) bool {
+		return asserterErrors[i].Code < asserterErrors[j].Code
+	})
+
+	for i, networkError := range networkErrors {
+		asserterError := asserterErrors[i]
+		if !reflect.DeepEqual(networkError, asserterError) {
+			return fmt.Errorf(
+				"/network/options / asserter-configuration-file error mismatch %v %v",
+				networkError, asserterError,
+			)
```

Comment:
for reporting differences in these verification methods, I'd recommend seeing if there is an easy way to give the user more insight into the mismatch....like these are the new errors we did not expect...or simply pretty print the two slices in the error message to help them debug.